data/dataset.py

```diff
@@ -32,35 +35,40 @@ def __init__(
             n_velocity_bins=n_velocity_bins,
         )
 
-        self.tokenizer_src = Tokenizer(keys=["pitch", "dstart_bin", "duration_bin", "velocity_bin"])
-        self.tokenizer_tgt = VelocityTokenizer()
+        self.tokenizer_src = src_tokenizer
+        self.tokenizer_tgt = tgt_tokenizer
+
+        if samples_path is None:
+            self.samples_path = (
+                f"tmp/datasets/samples-"
+                f"{self.quantizer.n_dstart_bins}-"
+                f"{self.quantizer.n_duration_bins}-"
+                f"{self.quantizer.n_velocity_bins}-"
+                f"{self.split}-"
+                f"{self.tokenizer_src.keys[0]}-{self.tokenizer_src.keys[1]}-vs-"
+                f"{self.tokenizer_tgt.keys[0]}-{self.tokenizer_tgt.keys[1]}.pt"
+            )
```

Comment:
Hmmmm but when you say outside the dataset, you mean, for example, in train.py file? Because in current approach I save a sliced dataset and tokeized samples to pass to the model. The way you are suggesting I would be saving and loading a pandas Dataframe, which I would have to load into the TokenizedMidiDataset. Is that what you would want?

Reply:
I meant saving the whole `TokenizedMidiDataset` (`pd.read_pickle` works with any object, not only dataframes, I assumed the same is true for parquet, sorry for the confusion).

This should work in any place, the dataset loading pattern could look like this:

```python
dataset_config = {"n_dstart_bins": 3, ...}
tokenized_midi_dataset = load_cache_dataset(dataset_config)
```

This may get trickier to support any hf dataset name and split, but you can always include this info in the config 🤔 Let me know if this makes sense!